schema.graphql

```diff
@@ -107,6 +121,162 @@ type UnprocessedEntity @entity {
   block: Block
 }
 
+type Service @entity {
+  id: ID!
+  name: String!
+  computeUnitsPerRelay: BigInt!
+  owner: Account!
+  addServiceMsgs: [AddServiceMsg] @derivedFrom(field: "service")
+}
+
+type AppStakeMsg @entity {
+  id: ID!
+  stake: Coin!
+  transaction: Transaction!
+  block: Block!
+  application: Application! @index
+}
+
+type AppStakeMsgService @entity {
+  id: ID!
+  appStakeMsg: AppStakeMsg!
+  service: Service!
+}
+
+type AppUnstakeMsg @entity {
+  id: ID!
+  application: Application! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type DelegateToGatewayMsg @entity {
+  id: ID!
+  application: Application! @index
+  gateway: Gateway! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type UndelegateToGatewayMsg @entity {
+  id: ID!
+  application: Application! @index
+  gateway: Gateway! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type TransferApplicationMsg @entity {
+  id: ID!
+  sourceApplication: Application! @index
+  destinationApplication: Application! @index
+  transaction: Transaction!
+  block: Block!
+}
+
+type Application @entity {
+  # id is the address
+  id: ID!
+  account: Account!
+  stake: Coin!
+  # 0 = Staked, 1 = Unstaking, 2 = Unstaked
+  status: Int!
```

Comment:
Again, I would push for a graphql enum here instead and if that's not feasible for whatever reason, then a [typescript enum](https://www.typescriptlang.org/docs/handbook/enums.html) instead.